### DISCUSSION TITLE
Python SDK for Experiment Management

### DISCUSSION THREAD

**Discussion by imirzadeh** [gridai/gridai#49] [Ideas]
## Disclaimer
I am very new to grid.ai and I might have got some concepts wrong.

## Idea
Currently, the dashboard is tightly coupled with Experiments/Runs.
However, I guess grid.ai extracts metrics from the runs by itself. Wouldn't it be helpful to use an SDK for reporting metrics/logging, and so on to the grid.ai dashboard?

I don't know if you are planning to have an experiment management system (like wandb & comet.ml) or not.
If this the plan, I guess it makes sense to have the ability to directly report some metrics/stats.

## More Context
An external API for creating/managing runs can give more power to the users. For example, users can use their own mechanisms for parameter sweeping (e.g., use Bayesian Optimization by their favorite library) and create many runs by themselves.

**Comment by pchitale1** [comment]
Thanks for the suggestion! We are looking into it
